src/lobe/Signature.py

```diff
@@ -1,24 +1,39 @@
 from __future__ import annotations
 import os
 import json
+import pathlib
 from typing import Tuple
 
 def load(model_path: str) -> Signature:
+    """
+    Loads the Signature for the given PATH or PATH_AND_FILENAME provided:
+    - Use PATH When: Using Lobe-Python in its default config, the Signature and TensorFlow (and TFLite) models are expected to be 
+        in one folder by themselves. Additional models may exist, but they too are expected to be in their own folders.
+    - Use PATH_AND_FILENAME When: Using Lobe-Python with multiple TensorFlow (and TFLite) models in the same folder, with
+        the Signature and Model files named uniquely. This allows you to store multiple TensorFlow/TFLite models and signatures in the same folder.
+    """
     model_path_real = os.path.realpath(os.path.expanduser(model_path))
-    if not os.path.isdir(model_path_real):
-        raise ValueError(f"Model directory does not exist: {model_path}")
-
-    signature_path = os.path.join(model_path_real, "signature.json")
-    if not os.path.isfile(signature_path):
-        raise ValueError(f"signature.json file not found at path: {model_path}")
 
-    return Signature(model_path_real)
+    #This could be a full_path to the signature file
+    if not os.path.isdir(model_path_real):
```

Comment:
Instead of checking for not dir, I would check for os.path.isfile() here, and make the else an elif os.path.isdir(), and then have the else raise a value error saying not a valid signature file or model directory